test/transform/forEachTest.js

```diff
@@ -0,0 +1,328 @@
+          'for (let i=0; i < array.length; i++) {\n' +
+          '  let item = array[i];\n' +
+          '  console.log(item, i);\n' +
+          '}'
+        ).toReturn(
+          'array.forEach((item, i) => {\n' +
+          '  console.log(item, i);\n' +
+          '});'
+        );
+      });
+
+      it('inside a function in loop body', () => {
+        expectTransform(
+          'for (let i=0; i < array.length; i++) {\n' +
+          '  let item = array[i];\n' +
+          '  callback(function(){ return i; });\n' +
+          '}'
+        ).toReturn(
+          'array.forEach((item, i) => {\n' +
+          '  callback(function(){ return i; });\n' +
+          '});'
+        );
+      });
+    });
+
+    describe('should transform array items used inside the body', () => {
+      it('for simple loops', () => {
+        expectTransform(
+          'for (let i = 0; i < xs.length; i++) {\n' +
+          '  const x = xs[i];\n' +
+          '  console.log(xs[i]);\n' +
+          '}'
+        ).toReturn(
+          'xs.forEach(x => {\n' +
+          '  console.log(x);\n' +
+          '});'
+        );
+      });
+
+      it('for nested loops', () => {
+        expectTransform(
+          'for (let i = 0; i < xs.length; i++) {\n' +
+          '  const x = xs[i];\n' +
+          '  console.log(xs[i]);\n' +
+          '  for (let j = 0; j < xs[i].length; j++) {\n' +
+          '    const y = xs[i][j];\n' +
+          '    console.log(xs[i][j]);\n' +
+          '  }\n' +
+          '}'
+        ).toReturn(
+          'xs.forEach(x => {\n' +
+          '  console.log(x);\n' +
+          '\n' +
+          '  x.forEach(y => {\n' +
+          '    console.log(y);\n' +
+          '  });\n' +
+          '});'
+        );
+      });
+    });
+
+    describe('should not transform when loop contains a ', () => {
+      it('break statement', () => {
+        expectNoChange(
+          'for (let i=0; i < array.length; i++) {\n' +
+          '  var item = array[i];\n' +
+          '  if (item == 2) {\n' +
+          '    break;\n' +
+          '  }\n' +
+          '}'
+        ).withWarnings([
+          {line: 1, msg: 'Break statement used in for-loop body', type: 'for-each'}
+        ]);
+      });
+
+      it('continue statement', () => {
+        expectNoChange(
+          'for (let i=0; i < array.length; i++) {\n' +
+          '  var item = array[i];\n' +
+          '  continue;\n' +
+          '}'
+        ).withWarnings([
+          {line: 1, msg: 'Continue statement used in for-loop body', type: 'for-each'}
+        ]);
+      });
+
+      it('when loop index variable defined with var', () => {
+        expectNoChange(
+          'for (var i=0; i < array.length; i++) {\n' +
+          '  let item = array[i];\n' +
+          '  console.log(item);\n' +
+          '}'
+        ).withWarnings([
+          {line: 1, msg: 'Only for-loops with let/const can be tranformed (use let transform first)', type: 'for-each'}
+        ]);
+      });
```

Comment:
What I meant was, that we should also have a test for this case (to be sure that we ignore loops that contain `var`):

```js
for (let i=0; i < array.length; i++) {
  var item = array[i]; // Here
  console.log(item);
}
```